### DISCUSSION TITLE
Module Listing

### DISCUSSION THREAD

**Discussion by PerryCameron** [wiverson/jtoolprovider-plugin#18] [Ideas]
I figured out that you could see what modules had been installed by looking into the package contents of your app and going to Contents/Runtime/Contents/Home/legal and there would be a directory for each module. I think it might be nice when doing "mvn clean install" under your tool jtoolprovider-plugin:1.0.34:java-tool that it would log each jvm.module to be installed under [INFO]. Just an idea.

I have one last problem, and it is probably a common one. When I try to make an Http request to a rest server I get:
[javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure, javax.net.ssl.SSLHandshakeException

I assumed that since it worked in the IDE but not in the jar it must be a missing module, so I added java.net.http. I was sure that would do it, but no dice.  I see that the SSL suite is in the java.base module so I should have those. Do you have any idea what I may be missing?

**Comment by wiverson** [comment]
Usually if there is a missing module it shows up as something more obvious, like a ClassDefNotFound or similar. handshake_failure makes me think there is something else going on with the connection. You might try adding more detail to the try block wrapping the exception to get more detail.

You can try running it from the command line with mvn javafx:run and see what you get there, you can also try attaching a debugger and see what's happening. There's a lot of weird stuff going on that could be affecting handshake (e.g. https://www.baeldung.com/java-ssl-handshake-failures and https://www.baeldung.com/java-keystore).

That said, it could be another one of these modules: https://docs.oracle.com/javase/9/security/java-security-overview1.htm#JSSEC-GUID-65C96219-B2AB-4205-808E-5B41CB2AD694 - can try adding them and see if that helps. If it's a cert looking for a specific security provider that's in one of those modules that could do it and you might only get that from the wrapped exception.

**Reply by PerryCameron** [reply]
I never wrote how I solved this for future readers. It was in fact missing a module.

jdk.crypto.ec

I found the answer here.
https://stackoverflow.com/questions/68896945/javax-net-ssl-sslhandshakeexception-happening-when-using-custom-java-runtime-ima
